src/mug.gleam

```diff
@@ -109,13 +177,79 @@ pub type ConnectionOptions {
     /// will also return a timeout, even if this timeout value has not been
     /// reached yet.
     timeout: Int,
+    /// TLS options
+    tls_opts: TlsConnectionOptions,
+  )
+}
+
+pub type TlsConnectionOptions {
+  /// Do not use TLS, use plain TCP
+  NoTls
+  /// Start with a TLS connection
+  UseTls(TlsOptions)
+}
+
+/// Configuration for TLS connections.
+pub type TlsOptions {
+  TlsOptions(verification_method: TlsVerificationMethod)
+}
+
+/// The certificates to use 
+pub type TlsVerificationMethod {
+  /// Uses these CA certificates and regular certificates to verify the server's certificate.
+  ///
+  /// The `use_system_cacerts` option makes mug use the system's CA certificates, which are 
+  /// usually set to a group of competent CAs who sign most of the web's certificates.  
+  /// You may specifiy your own CA certificates with the `cacerts` option, and custom
+  /// certificates and their keys with the `certs_keys` option.
+  ///
+  /// Note that specifying a PEM encoded CA certificate file will result in the system CA
+  /// certificates not being used. This is because of how the underlying [erlang implementation](https://www.erlang.org/doc/apps/ssl/ssl.html#t:client_option/0)
+  /// works. System CA certificates are passed as DER-encoded certificates, and DER encoded
+  /// certs override PEM-encoded ones. Therefore, if you wish to use a PEM encoded CA cert along
+  /// with the system's CA, you should decode the PEM into a DER using (`public_key:pem_decode`)[https://www.erlang.org/doc/apps/public_key/public_key.html#pem-api]
+  /// and use the DerEncodedCaCertificates variant instead.
+  Certificates(
+    use_system_cacerts: Bool,
+    cacerts: Option(CaCertificates),
+    certs_keys: List(CertsKeys),
```

Comment:
```suggestion
    certificate_keys: List(CertsKeys),
```

Reply:
No we should not.

Reply:
Should not the type name be CertificateKeys also then? Or even CertificatesKeys?

Reply:
i've updated the field to be `certificates_keys`, and the type to be `CertificatesKeys`